airflow/jobs/backfill_job.py

```diff
@@ -645,16 +645,14 @@ def tabulate_ti_keys_set(set_ti_keys: Set[TaskInstanceKey]) -> str:
             # Sorting by execution date first
             sorted_ti_keys = sorted(
                 set_ti_keys,
-                key=lambda ti_key: (ti_key.execution_date, ti_key.dag_id, ti_key.task_id, ti_key.try_number),
+                key=lambda ti_key: (ti_key.run_id, ti_key.dag_id, ti_key.task_id, ti_key.try_number),
```

Comment:
This would change the order that errors are printed only so it's not "significant" to anything.

(The session handling here is all over the shop, and if the TI is detached from the session and hasn't yet loaded `ti.dag_run` then you get a DetachedObject exception)

Reply:
https://docs.sqlalchemy.org/en/13/orm/loading_relationships.html?highlight=joinedload#sqlalchemy.orm.contains_eager may be useful in various places (`joinedload` would join against the table _twice_)

Reply:
in that case L645 needs to be updated

Reply:
Already updated in follow on pr